File: Java/Gate-four/LevelFiveFunctionTest.java
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class LevelFiveFunctionTest      {

@Test

public void testSortPrimeNumbers()       {

    int[] input = {5, 9, 3, 6, 2};

    int[] expected = {5,3, 2};

    int[] actual = LevelFiveFunction.sortArrayPrime(input);
    
    assertArrayEquals (expected, actual);


       }


    @Test

public void testNegativeNumbersReplacement()       {

    int[] input = {5, -9, 3, -6, 2, -11};

    int[] expected = {5, 0, 3, 0, 2, 0};

    int[] actual = LevelFiveFunction.replaceNagativeNumbers(input);
    
    assertArrayEquals (expected, actual);


       }

 @Test
public void testRearrangeArray()       {

    int[] input = {5, 0, 3, 0, 2, 0};

    int[] expected = {5, 3, 2, 0, 0, 0};

    int[] actual = LevelFiveFunction.rearrangeArray(input);
    
    assertArrayEquals (expected, actual);


       }




@Test
public void testFindEvenNumbers()       {

    int[] input = {45, 60, 3, 10, 9, 22};

    int[] expected = {60, 10, 22};

    int[] actual = LevelFiveFunction.evenArray(input);
    
    assertArrayEquals (expected, actual);


       }

@Test
public void testFindOddNumbers()       {

    int[] input = {45, 60, 3, 10, 9, 22};

    int[] expected = {45, 3, 9};

    int[] actual = LevelFiveFunction.oddArray(input);
    
    assertArrayEquals (expected, actual);


       }


@Test
public void testIsPalindromeArray()       {

    int[] input = {45, 0, 8, 0, 45};
    
    assertTrue(LevelFiveFunction.findPalindromeArray(input));

}

@Test
public void testIsNotPalindromeArray()       {

    int[] input = {1, 2, 3, 4, 5};
    
    assertFalse(LevelFiveFunction.findPalindromeArray(input));

}

@Test
public void testFindPerfectSquares()       {

    int[] input = {4, 7, 9, 10, 16, 18};

    int[] expected = {4, 9, 16};

    int[] actual = LevelFiveFunction.findPerfectSquares(input);
    
    assertArrayEquals (expected, actual);


       }

















    }

File: Python/Gate-four/level_five_function_test.py
import unittest
from level_five_function import *

class TestLevelFiveFunction(unittest.TestCase):

    def test_negative_numbers_replacement(self):
        input_list = [5, -9, 3, -6, 2, -11]
        expected = [5, 0, 3, 0, 2, 0]
        actual = replace_negative_numbers(input_list)
        self.assertEqual(expected, actual)

    def test_rearrange_array(self):
        input_list = [5, 0, 3, 0, 2, 0]
        expected = [5, 3, 2, 0, 0, 0]
        actual = rearrange_array(input_list)
        self.assertEqual(expected, actual)

    def test_find_even_numbers(self):
        input_list = [45, 60, 3, 10, 9, 22]
        expected = [60, 10, 22]
        actual = even_array(input_list)
        self.assertEqual(expected, actual)

    def test_find_odd_numbers(self):
        input_list = [45, 60, 3, 10, 9, 22]
        expected = [45, 3, 9]
        actual = odd_array(input_list)
        self.assertEqual(expected, actual)
